FreeRTOS/FreeRTOS-Kernel
# Implement prvYieldCore with macro

Description
-----------
* Implement prvYieldCore with macro
* Remove the portCHECK_IF_IN_ISR macro check. It is not required in SMP now
* Not checking current task is scheduled to yield in prvYieldCore. When a task entering the critical section, the logic in [prvCheckForRunStateChange ](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/2f94b181a2f049ec342deba0927bed51f7174ab0/tasks.c#L704) ensures that the run state of current task is not `taskTASK_SCHEDULED_TO_YIELD`

Test Steps
-----------
N/A

Checklist:
----------



Related Issue
-----------
Address #750 comment in line 760


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
